pyaedt/modules/Boundary.py

```diff
@@ -4520,3 +4521,232 @@ def _create_boundary(bound):
             raise Exception
     except Exception:  # pragma: no cover
         return None
+
+
+class BoundaryDictionary:
+    """
+    Handles Icepak transient and temperature-dependent boundary condition assignments.
+
+    Parameters
+    ----------
+    assignment_type : str
+        Type of assignment represented by the class. Options are `"Temp Dep"``
+        and ``"Transient"``.
+    function_type : str
+        Variation function to assign. If ``assignment_type=="Temp Dep"``,
+        the function can only be ``"Piecewise Linear"``. Otherwise, the function can be
+        ``"Exponential"``, ``"Linear"``, ``"Piecewise Linear"``, ``"Power Law"``,
+        ``"Sinusoidal"``, and ``"Square Wave"``.
+    """
+
+    def __init__(self, assignment_type, function_type):
+        if assignment_type not in ["Temp Dep", "Transient"]:  # pragma : no cover
+            raise AttributeError("The argument {} for ``assignment_type`` is not valid.".format(assignment_type))
+        if assignment_type == "Temp Dep" and function_type != "Piecewise Linear":  # pragma : no cover
+            raise AttributeError(
+                "Temperature dependent assignments only support"
+                ' ``"Piecewise Linear"`` as ``function_type`` argument.'.format(assignment_type)
+            )
```

Comment:
Same remark as previous comments

Reply:
How should I handle this one? I cannot return False as I would get a `TypeError`. These lines are never run (that's why they're not covered by tests) unless the user tries to manually initialize the class which would lead literally to nowhere (the only property here, `props`, uses `_parse_value` which is only abstract in the base class).

Basically, these lines are there just as sanity check for future implementations (if new conditions will be added to native APIs). As far as I'm concerned, we can leave those. But if we really don't want those error raises, I can delete them. What do you prefer?

Reply:
You are right, my comment was not clear enough and I apologize. I wanted to pinpoint the use of the `logger.error` but you can leave it as it is !